src/contract.ts

```diff
@@ -45,14 +44,42 @@ export class Contract {
             Object.defineProperty(this, methodName, {
                 writable: false,
                 enumerable: true,
-                value: nameFunction(methodName, async (args: object = {}, gas?: BN, amount?: BN, ...ignored) => {
-                    if (ignored.length || !(isObject(args) || isUint8Array(args))) {
-                        throw new PositionalArgsError();
+                value: nameFunction(
+                    methodName,
+                    async (
+                        args: object = {},
+                        optionsOrGas?: ChangeMethodOptions | BN,
+                        deposit?: BN,
+                        ...ignored
+                    ) => {
+                        const options = isObject(optionsOrGas)
```

Comment:
seems like can be a problem in case when `BN` is used for gas (isn't it an object?)